tests/integration-tests/tests/ultraserver/test_gb200/test_gb200/90-nvidia-imex.prolog.sh

```diff
@@ -91,14 +91,40 @@ function write_file() {
     return 1 # Not Updated
   fi
 
-  touch "${_lock_file}"
+  # Try to acquire lock with timeout
+  (
+      if ! flock -x -w ${_lock_timeout_seconds} 200; then
+        # If timeout, assume deadlock and try to recover
+        info "Lock timeout after ${_lock_timeout_seconds}s, attempting deadlock recovery"
+        exit 1
+      fi
+      echo "${_content}" > "${_file}"
+  ) 200>"${_lock_file}"
+
+  local _lock_result=$?
+
+  if [[ ${_lock_result} -eq 0 ]]; then
+    return 0 # Updated successfully
+  fi
+
+  # Deadlock recovery: remove stale lock file and retry once
+  error "Potential deadlock detected for ${_file}, attempting recovery"
+  rm -f "${_lock_file}"
+  sleep 1  # Brief pause to avoid race conditions
 
   (
-      flock -x -w ${_lock_timeout_seconds} 200 || error_exit "Cannot acquire lock to write to ${_file}"
+      if ! flock -x -w 10 200; then
+        exit 1
+      fi
       echo "${_content}" > "${_file}"
   ) 200>"${_lock_file}"
 
-  return 0 # Updated
+  if [[ $? -eq 0 ]]; then
```

Comment:
Lets create another functions as we keep repeating the same logic of writing in a lock file and then checking if the sub-block was succesful or not and writing different log lines

Reply:
Got it. So as discussed we will address this in next iteration